homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -434,35 +434,33 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         if DATA_KEY not in hass.data:
             hass.data[DATA_KEY] = {}
```

Comment:
We don't seem to use `DATA_KEY` in hass.data anymore.

How will the services access the entities now? Please review the entities reference and consider changing the variable names to distinguish between entities filtered in the service handler and entities added to Home Assistant.

Reply:
I will have a look. @bieniu  could you test services?

Reply:
Note that this issue will be solved naturally by moving to use the entity service helper.

Reply:
Then we just need a work-a-round for now.

Reply:
I think the service handler can use the `entities` reference for the entities that are added to Home Assistant since the platform will only be set up once now, instead of multiple times. But for clarity it would then be good change the name of the variable in the service handler for the filtered entities.

Reply:
Will the supplies solution work. We don't have the `host` variable, we have `unique_id` which should be unique to a device.
```python
hass.data[DATA_KEY][unique_id] = entity
```

Reply:
Service call seems to be working normally. `fan_set_led_brightness` changes `led_brightness` to `0`:
```
2021-08-10 11:48:56 DEBUG (MainThread) [homeassistant.components.xiaomi_miio.device] Response received from miio device: [{'did': 'led_brightness', 'siid': 6, 'piid': 1, 'code': 0}]
2021-08-10 11:49:07 DEBUG (MainThread) [homeassistant.components.xiaomi_miio] Got new state: <AirPurifierMiotStatus aqi=1 average_aqi=50 buzzer=False buzzer_volume=None child_lock=False fan_level=1 favorite_level=5 filter_hours_used=101 filter_life_remaining=97 filter_rfid_product_id=0:0:31:31 filter_rfid_tag=80:6c:ba:b2:fc:d8:4 filter_type=FilterType.Regular humidity=47 is_on=True led=True led_brightness=LedBrightness.Off mode=OperationMode.Silent motor_speed=389 power=on purify_volume=10555 temperature=25.8 use_time=353400>
```

Reply:
> 
> 
> I think the service handler can use the `entities` reference for the entities that are added to Home Assistant since the platform will only be set up once now, instead of multiple times. But for clarity it would then be good change the name of the variable in the service handler for the filtered entities.

Is there suggestion for a name change?

Reply:
Eg `filtered_entities`.

Reply:
See last commit, is this what you mean?